cotix/_collision_resolution.py

```diff
@@ -0,0 +1,79 @@
+from jax import numpy as jnp
+from jaxtyping import Array, Float
+
+from cotix._bodies import AbstractBody, Ball
+from cotix._geometry_utils import perpendicular_vector
+
+
+def _split_bodies(
+    body1: AbstractBody, body2: AbstractBody, epa_vector: Float[Array, "2"]
+) -> (AbstractBody, AbstractBody):
+    # lets apply translation to both bodies, taking their mass into account
+    epsilon = 1e-3
+    body1 = body1.set_position(
+        body1.position
+        + (1 + epsilon) * epa_vector * (body2.mass / (body1.mass + body2.mass))
+    )
+    body2 = body2.set_position(
+        body2.position
+        - (1 + epsilon) * epa_vector * (body1.mass / (body1.mass + body2.mass))
+    )
```

Comment:
I think we need the resolution with smaller size or equal size than EPA. Like, we are alright with bodies being a bit inside each other, but not alright if they start oscillate like crazy. The second scenario might happen when you have ```1+epsilon```

Reply:
tbh i added epsilon only when the tests weren't always passing. lets discuss tmrw in-depth, I have quite some questions

Reply:
alright, will multiply by `1/5` or something rather than `(1 + epsilon)`

Reply:
i thought again about the 1/5, and the walls example, why cant we just make the walls immovable? and not update their position. with the gradual position change, we will also want a gradual velocity change, right? or do we just let the bodies flow out of each other after the initial speed change?
can still remove the epsilon though